pkg/checks/checks.go

```diff
@@ -54,68 +56,193 @@ func WaitForHTTPStatusOk(ctx context.Context, transport http.RoundTripper, urlto
 	}, ctx.Done())
 }
 
-func checkNamespace(namespace string) bool {
-	if namespace == "default" {
-		return true
-	}
+type Node struct {
+	Ready bool
+	Name  string
+}
 
-	whitelistedNamespaces := []string{
-		"openshift-",
-		"kube-",
-	}
-	for _, ns := range whitelistedNamespaces {
-		if strings.HasPrefix(namespace, ns) {
-			return true
+func GenerateNodeMap(cs *acsapi.OpenShiftManagedCluster) map[string][]Node {
+	nodes := make(map[string][]Node)
+	for _, app := range cs.Properties.AgentPoolProfiles {
+		nodes[string(app.Role)] = []Node{}
+		for i := 0; i < app.Count; i++ {
+			n := Node{Name: fmt.Sprintf("%s-%06d", app.Role, i)}
+			nodes[string(app.Role)] = append(nodes[string(app.Role)], n)
 		}
 	}
-	return false
+	return nodes
 }
 
-// WaitForInfraServices verify daemonsets, statefulsets
-func WaitForInfraServices(ctx context.Context, appclient *appclient.AppsV1Client) error {
-	log.Info("checking infrastructure service health")
-out:
-	for {
-		_, err := appclient.Deployments("openshift-web-console").Get("webconsole", metav1.GetOptions{})
-		switch {
-		case err == nil:
-			break out
-		case kerrors.IsNotFound(err):
-		default:
-			return err
-		}
-		select {
-		case <-time.After(2 * time.Second):
-		case <-ctx.Done():
-			return ctx.Err()
+func isNodeReady(nodeName string, kc *kubernetes.Clientset) (bool, error) {
+	n, err := kc.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
```

Comment:
@kargakis, I believe this is the call that timed out.  Is this an issue with the way we handle the returning err?  Should we do something like this? https://github.com/openshift/openshift-azure/blob/master/pkg/checks/checks.go#L34-L49